src/libraries/System.Security.Cryptography.Cose/tests/CoseTestHelpers.cs

```diff
@@ -144,54 +168,79 @@ private static CoseHeaderMap AssertSign1Headers(CborReader reader, int expectedH
             return headers;
         }
 
-        internal static readonly Dictionary<ECDsaAlgorithm, ECDsa> ECDsaKeys = CreateECDsaKeys(true);
+        private static ECParameters _ec256Parameters = CreateECParameters("nistP256", "usWxHK2PmfnHKwXPS54m0kTcGJ90UiglWiGahtagnv8", "IBOL-C3BttVivg-lSreASjpkttcsz-1rb7btKLv8EX4", "V8kgd2ZBRuh2dgyVINBUqpPDr7BOMGcF22CQMIUHtNM");
+        private static ECParameters _ec384Parameters = CreateECParameters("nistP384", "kTJyP2KSsBBhnb4kjWmMF7WHVsY55xUPgb7k64rDcjatChoZ1nvjKmYmPh5STRKc", "mM0weMVU2DKsYDxDJkEP9hZiRZtB8fPfXbzINZj_fF7YQRynNWedHEyzAJOX2e8s", "ok3Nq97AXlpEusO7jIy1FZATlBP9PNReMU7DWbkLQ5dU90snHuuHVDjEPmtV0fTo");
+        private static ECParameters _ec512Parameters = CreateECParameters("nistP521", "AHKZLLOsCOzz5cY97ewNUajB957y-C-U88c3v13nmGZx6sYl_oJXu9A5RkTKqjqvjyekWF-7ytDyRXYgCF5cj0Kt", "AdymlHvOiLxXkEhayXQnNCvDX4h9htZaCJN34kfmC6pV5OhQHiraVySsUdaQkAgDPrwQrJmbnX9cwlGfP-HqHZR1", "AAhRON2r9cqXX1hg-RoI6R1tX5p2rUAYdmpHZoC1XNM56KtscrX6zbKipQrCW9CGZH3T4ubpnoTKLDYJ_fF3_rJt");
 
-        internal static readonly Dictionary<ECDsaAlgorithm, ECDsa> ECDsaKeysWithoutPrivateKey = CreateECDsaKeys(false);
+        [ThreadStatic]
+        private static ECDsa _es256 = CreateECDsa(_ec256Parameters, true);
+        [ThreadStatic]
+        private static ECDsa _es384 = CreateECDsa(_ec384Parameters, true);
+        [ThreadStatic]
+        private static ECDsa _es512 = CreateECDsa(_ec512Parameters, true);
 
-        internal static readonly RSA RSAKey = CreateRSA(true);
+        [ThreadStatic]
+        private static ECDsa _es256WithoutPrivateKey = CreateECDsa(_ec256Parameters, false);
+        [ThreadStatic]
+        private static ECDsa _es384WithoutPrivateKey = CreateECDsa(_ec384Parameters, false);
+        [ThreadStatic]
+        private static ECDsa _es512WithoutPrivateKey = CreateECDsa(_ec512Parameters, false);
+
+        internal static ECDsa DefaultKey { get; } = _es256;
+        internal static HashAlgorithmName DefaultHash { get; } = GetHashAlgorithmNameFromCoseAlgorithm((int)ECDsaAlgorithm.ES256);
 
+        internal static readonly Dictionary<ECDsaAlgorithm, ECDsa> ECDsaKeys = CreateECDsaDictionary(true);
+        internal static readonly Dictionary<ECDsaAlgorithm, ECDsa> ECDsaKeysWithoutPrivateKey = CreateECDsaDictionary(false);
```

Comment:
These dictionaries are also undoing the `[ThreadStatic]` and pulling the key objects back to one object shared across all threads.

Reply:
Is the fix here to apply the same pattern `t_es256 ??= CreateECDsa(_ec256Parameters, true)` to the returned KVPs in `CreateECDsaDictionary()`?

Reply:
I guess not, we either make the dictionaries to be properties (which will create them every time it gets called) or make them `ThreadStatic` as well.

Reply:
For what you're trying to achieve, you'd need the dictionary to be ThreadStatic.  Or to make your own item with an indexer that returns the thread-static fields (probably accessed via the properties so you don't have to repeat the initializers)

Reply:
Done (or an attempt) in https://github.com/dotnet/runtime/pull/64461/commits/cefe68610e38ecefe10b8a12e6e0f701c12a864f.